tests/transports/test_all_plugins.py

```diff
@@ -91,31 +90,64 @@ def test_is_open(custom_transport):
     assert not custom_transport.is_open
 
 
+# Add this function near the top of the file after the imports
+def safe_unpack_archive(filename, extract_dir):
+    """Unpack an archive with safe filtering to avoid warnings."""
+    if (
+        filename.endswith('.tar')
+        or filename.endswith('.tar.gz')
+        or filename.endswith('.tar.bz2')
+        or filename.endswith('.tar.xz')
+    ):
```

Comment:
@khsrali this seems changes from your PR, correct? Please make sure these are cleaned before merge.